soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -209,12 +226,57 @@ void func_809DF730(EnCow* this, GlobalContext* globalCtx) {
     }
 }
 
+bool EnCow_IsSecondCowInTower(EnCow* this, GlobalContext* globalCtx) {
+    return 
+        gSaveContext.n64ddFlag && 
+        globalCtx->sceneNum == SCENE_SOUKO && 
+        (this->actor.world.pos.x + this->actor.world.pos.z == -173 || this->actor.world.pos.x + this->actor.world.pos.z == -72);
+}
+
+bool EnCow_IsSecondCowInStable(EnCow* this, GlobalContext* globalCtx) {
+    return 
+        gSaveContext.n64ddFlag && 
+        globalCtx->sceneNum == SCENE_MALON_STABLE && 
+        (this->actor.world.pos.x + this->actor.world.pos.z == -257 || this->actor.world.pos.x + this->actor.world.pos.z == -138);
+}
+
+bool EnCow_IsSecondCowInGrotto(EnCow* this, GlobalContext* globalCtx) {
+    return gSaveContext.n64ddFlag && globalCtx->sceneNum == SCENE_KAKUSIANA && this->actor.world.pos.x + this->actor.world.pos.z == 3194;
+}
+
+bool EnCow_IsSecondCow(EnCow* this, GlobalContext* globalCtx) {
+    return EnCow_IsSecondCowInTower(this, globalCtx) || EnCow_IsSecondCowInStable(this, globalCtx) || EnCow_IsSecondCowInGrotto(this, globalCtx);
+}
+
+GetItemID EnCow_GetItemFromCow(EnCow* this, GlobalContext* globalCtx, bool setFlag) {
+    GetItemID itemId = ITEM_NONE;
+
+    if (Inventory_HasEmptyBottle()) {
+        return GI_MILK;
+    }
```

Comment:
it seems like having this here will make it so milk overrides the rando check if the player has an empty bottle

Reply:
This runs first, then the rando check after so the rando would take priority, but I’ll move this to an else to make it more readable

Reply:
Oops actually you're right, this was meant to be `itemId = GI_MILK;` not return, regardless It's moving 👍